actions.go

```diff
@@ -156,25 +199,35 @@ func (c *Action) Fatalf(msg string, args ...interface{}) {
 // Warningf prints a warning-level message. The arguments follow the standard
 // Printf arguments.
 func (c *Action) Warningf(msg string, args ...interface{}) {
-	fmt.Fprintf(c.w, warningFmt, c.fields, fmt.Sprintf(msg, args...))
+	// ::warning <c.fields>::<msg, args>
+	c.IssueCommand(&Command{
+		Name:       warningCmd,
+		Message:    fmt.Sprintf(msg, args...),
+		Properties: c.fields,
+	})
 }
 
 // WithFieldsSlice includes the provided fields in log output. "f" must be a
 // slice of k=v pairs. The given slice will be sorted.
 func (c *Action) WithFieldsSlice(f []string) *Action {
-	sort.Strings(f)
-	return &Action{
-		w:      c.w,
-		fields: " " + strings.Join(f, ","),
+	m := make(CommandProperties)
+	for _, s := range f {
+		pair := strings.SplitN(s, "=", 2)
+		if len(pair) < 2 {
+			panic(fmt.Sprintf("%q is not a proper k=v pair!", s))
+		}
+
+		m[pair[0]] = pair[1]
```

Comment:
So, it's worth noting that this is changing the order in which the values appear. Maps are inherently unordered in Go, and field ordering _might_ matter in some cases.

Reply:
Sorry, forgot to mention in the comments. I took care of that in [CommandProperties.String()](https://github.com/sethvargo/go-githubactions/pull/10/files#diff-4d7d1923693fc5ce892add2ea2907a744e77ea0b50c1939ccc5067cb48a466a3R18-R28).

Reply:
I'm worried about a situation where someone defined:

```
FOO=bar
FOO=zap
```

it's technically value for _both_ of those to be in the environment.

Reply:
I'm sorry! I am not following? If the same key is used with two different values, shouldn't one value override the other?

Reply:
The POSIX environ (`os.Environ()` in Go). It's legitimate and valid to have the same key with two different values _and_ they are interpreted differently in some contexts: https://wiki.sei.cmu.edu/confluence/display/c/ENV02-C.+Beware+of+multiple+environment+variables+with+the+same+effective+name

Reply:
@sethvargo Sorry, didn't see your other comment.

> I'm not sure how I feel about converting the slice to a map. I'd rather convert the map to a slice.

If your concern is limited to this, the TypeScript toolkit also uses a JS object (analogous to maps) to hold properties. I noticed these bits at the end of all the changes so I didn't pour that much thought into it.

Reply:
Ignore my last (now deleted comment) about overridden duplicates. These aren't environment variables but extra parameters required by the commands.

```
::set-env <key>=<val>::<env_val>
```

Premature apologies if I stop making sense, it's past midnight here! xD

Reply:
@sethvargo I tried running a GitHub workflow to verify the behaviour on duplicate `key` in `key=val` pairs.

1. Running `echo "::warning file=actions.go,file=command.go,line=10,line=5,line=1::Test"` resulted in [this annotation](https://github.com/ashutoshgngwr/go-githubactions/actions/runs/308296353)
2. Runing `echo "::warning file=command.go,file=.github/workflows/duplicate-test.yaml,line=10,line=5,line=1::Test"` resulted in [this annotation](https://github.com/ashutoshgngwr/go-githubactions/actions/runs/308303211)

In both cases, last provided value is being used for duplicate keys.

Reply:
Okay thanks for digging.